### PR TITLE
feat: bump dex chart to support webhooks

### PR DESCRIPTION
**Pull request by palexster**
Bump dex chart to include the support for connectors and claims webhooks

https://d2iq.atlassian.net/browse/D2IQ-99457

### CODE REVIEW

File: stable/dex/templates/secret.yaml [hunk 1]
```diff
@@ -49,4 +49,20 @@ stringData:
...
+      redirectURIs:
+        - 'http://localhost:9009/auth/callback'
+      name: 'Example App'
+      secret: ZXhhbXBsZS1hcHAtc2VjcmV0
+    staticPasswords:
+      - email: "admin@example.com"
+        # bcrypt hash of the string "password": $(echo password | htpasswd -BinC 10 admin | cut -d: -f2)
+        hash: "$2a$10$2b2cU8CPhOTaGrs1HRQuAueS7JTT5ZHsHSzYiFPm1leZck7Mc8T4W"
+        username: "admin"
+        userID: "08a8684b-db88-4b73-90a9-3cd1661f5466"
```
Comment (mhrabovcin) [inline stable/dex/templates/secret.yaml:68]: Do we need this section?

File: stable/dex/values.yaml [hunk 2]
```diff
@@ -192,6 +192,37 @@ config:
...
+        - "email"
+        - "username"
+        - "preferred_username"
+      config:
+        url: "https://fulmicotone-dex-controller-webhook-service:18443/claims"
+        tlsRootCAFile: "/etc/dex/tls/client/ca.crt"
+        clientAuthentication:
+          clientKeyFile: "/etc/dex/tls/client/tls.key"
+          clientCertificateFile: "/etc/dex/tls/client/tls.crt"
+          ClientCAFile: "/etc/dex/tls/client/ca.crt"
```
Comment (mhrabovcin) [inline stable/dex/values.yaml:225]:
```suggestion
          clientCAFile: "/etc/dex/tls/client/ca.crt"
```

File: stable/dex/values.yaml [hunk 3]
```diff
@@ -192,6 +192,37 @@ config:
     tlsCert: /etc/dex/tls/grpc/server/tls.crt
     tlsKey: /etc/dex/tls/grpc/server/tls.key
     tlsClientCA: /etc/dex/tls/grpc/ca/tls.crt
+  connectorFiltersHooks:
```
Comment (mhrabovcin) [inline stable/dex/values.yaml:195]: Can we add this blocks as commented out?

File: stable/dex/templates/secret.yaml (hunk 1 above)
Comment (palexster) [reply stable/dex/templates/secret.yaml:68]: No, thanks. Some leftovers from testing.

File: stable/dex/values.yaml [hunk 4]
```diff
@@ -3,9 +3,9 @@
 # Declare name/value pairs to be passed into your templates.
 # name: value
 
-image: docker.io/mesosphere/dex
-imageTag: "v2.35.1-d2iq.1"
-imagePullPolicy: "IfNotPresent"
+-image: docker.io/mesosphere/dex
+-imageTag: "v2.37.0-d2iq-1"
```
Comment (mhrabovcin) [inline stable/dex/values.yaml:7]:
```suggestion
-imageTag: "v2.37.0-d2iq.1"
```

Did the tag format changed?

File: stable/dex/values.yaml (hunk 3 above)
Comment (palexster) [reply stable/dex/values.yaml:195]: Done in  030920e105693768c5a7cd40cfbe041732f64434

File: stable/dex/templates/secret.yaml (hunk 1 above)
Comment (palexster) [reply stable/dex/templates/secret.yaml]: 030920e105693768c5a7cd40cfbe041732f64434

**Review by mhrabovcin — APPROVED** [review]

**Review by ahmedElqutb — APPROVED** [review]

**Pull request merged**